core/src/main/java/com/novoda/espresso/TalkBackTestRule.java

```diff
@@ -0,0 +1,42 @@
+package com.novoda.espresso;
+
+import android.os.SystemClock;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class TalkBackTestRule implements TestRule {
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return new TalkBackStatement(base);
+    }
+
+    private static class TalkBackStatement extends Statement {
+
+        private static final int DEFAULT_DELAY_MILLIS = 1500;
+
+        private final TalkBackStateSettingRequester talkBackStateSettingRequester = new TalkBackStateSettingRequester(0);
+        private final Statement baseStatement;
+
+        TalkBackStatement(Statement baseStatement) {
+            this.baseStatement = baseStatement;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
+            talkBackStateSettingRequester.requestEnableTalkBack();
+            sleepToAllowTalkBackServiceToChangeState();
+
+            baseStatement.evaluate();
+
+            talkBackStateSettingRequester.requestDisableTalkBack();
+            sleepToAllowTalkBackServiceToChangeState();
+        }
+
+        private void sleepToAllowTalkBackServiceToChangeState() {
+            SystemClock.sleep(DEFAULT_DELAY_MILLIS);
```

Comment:
is there a way to check if talkback has been enabled? It'd be safer to sleep until it's enabled, instead of having a fixed `sleep`

Reply:
not specifically TalkBack - we can poll to see if there are _any_ spoken feedback services available. Could do this as a separate issue since this is existing behaviour :grin: just moved where the sleep was.

Reply:
true 👍